### ISSUE TITLE
app: Social Profile App

### ISSUE THREAD

**Issue by rsodre** [underware-gg/pistols-discord-bot#5]
To be able to tag Discord users when their Duelist receives a notification, like new Challenges (#4), we need to link their Discord and Pistols profiles.

This app is supposed to bare minimum, including a Discord login and a Pistols/Dojo wallet login.

For the sake of simplicity (and because the supported wallets for Pistols are TBD), instead of connecting a wallet, players can simply paste their duelist account address.

## Tasks

- [ ] Copy the [underware.gg](https://github.com/underware-gg/underware_gg) contents to `/social-app` as a starting point. It is a simple next.js app, with a single page already in Pistols styling.
- [ ] Use the [Discord SDK](https://github.com/discord/embedded-app-sdk) for login and anything else we require that it provides. [SDK Docs are here](https://discord.com/developers/docs/developer-tools/embedded-app-sdk).
- [ ] Create a **Login with Discord** button, using the Discord [OAuth2](https://discord.com/developers/docs/topics/oauth2) API
- [ ] When logged in, display a simple Dashboard with the Discord Id and profile picture
- [ ] The dashboard contains a simple form to edit the player's duelist address
- [ ] Store the duelist address at Supabase, using Discord User Id as key. We will provide the API keys and database schema.

## Notes

There are a few missing pieces here to make it work...

* Does the Discord API allow us to store custom information linked to our app?
* If not, is there any extension to do that?
* If not, is there a secure way to store make this Discord-Address link on-chain, on the Pistols Dojo world?
* The user address must be accessible from the bot!

If you have knowledge of the Discord API and/or have a solution for this, please reply explaining how would you do it.

**Comment by Dprof-in-tech** [comment]
hi @rsodre 

The discord api does not expressly allow us to store information but we can utilize a database system linked to our app to store the users pistols/Dojo wallet addresses and only authenticate them using the discord login id.

Then we can create a backend endpoint to retrieve this user address based on the discord login before making it available to the discord bot.

I think I can work on this.

**Comment by rsodre** [comment]
> The discord api does not expressly allow us to store information but we can utilize a database system linked to our app to store the users pistols/Dojo wallet addresses and only authenticate them using the discord login id.

you mean something like supabase?

**Comment by Dprof-in-tech** [comment]
> > The discord api does not expressly allow us to store information but we can utilize a database system linked to our app to store the users pistols/Dojo wallet addresses and only authenticate them using the discord login id.
> 
> you mean something like supabase?

yeah
supabase
firebase
mysql
postgresql

anyone works

**Comment by rsodre** [comment]
ok, will be assigning this to you.
I was going to use our [website](https://github.com/underware-gg/underware_gg) as a template since it is just a simple next.js app with some basic styling taken from pistols.

But I don't want to make you way wait, could you do that?
Copy it to `/social-app`

**Comment by rsodre** [comment]
About the database, if we can't store in on Discord, the best approach would be to store on-chain.
Will be reviewing it in a few hours

**Comment by Dprof-in-tech** [comment]
> About the database, if we can't store in on Discord, the best approach would be to store on-chain. Will be reviewing it in a few hours

ok then.
let's see how easy it is to store the data on chain and then retrieve it so the bot can access it

**Comment by Dprof-in-tech** [comment]
> ok, will be assigning this to you. I was going to use our [website](https://github.com/underware-gg/underware_gg) as a template since it is just a simple next.js app with some basic styling taken from pistols.
> 
> But I don't want to make you way wait, could you do that? Copy it to `/social-app`

I sure can do this.

**Comment by Dprof-in-tech** [comment]
I'm having some issues setting up tye discord bot on my.local machine.
Please is there a comprehensive guide on how to do this?

**Comment by rsodre** [comment]
The README here explains the setup process
https://github.com/underware-gg/pistols-discord-bot/tree/main/discord-bot

Let's use this server to configure and test it:
https://discord.gg/YVDqKCgU

**Comment by Dprof-in-tech** [comment]
Ok chief
Let me look at it real quick

**Comment by rsodre** [comment]
The main issue here is to link a Discord user with their duelist address.
A running bot is not required, it just needs to be configured.
You can create a Discord app on your Discord account to test it out. Once it is working and ready, we create our Discord app, with the same configurations and change the codes.

For user authentication:
https://discord.com/developers/docs/topics/oauth2

For storing the address, I was thinking we could use this, but it's not clear to me if that's what we need. I think not...
https://discord.com/developers/docs/resources/application-role-connection-metadata

Googling it everyone says the same as you said, we need a database.
If that's the case... ~~I need to extract some Dojo components from Pistols to use on this app and store it all on-chain~~, we'll setup a supabase database for it...

**Comment by Dprof-in-tech** [comment]
ok then , supabase it is

**Comment by rsodre** [comment]
I updated the issue description according to what we've been discussing

**Comment by Dprof-in-tech** [comment]
i've seen it and i'm moving on with the new description

**Comment by Dprof-in-tech** [comment]
i've been working on the login process but nextauth keeps returning an error. I'm trying to find an alternative to using nextauth

**Comment by Dprof-in-tech** [comment]
> i've been working on the login process but nextauth keeps returning an error. I'm trying to find an alternative to using nextauth

I've solved that and I've been able to retrieve the user details after they login. now to create the form for the duel wallet address

**Comment by Dprof-in-tech** [comment]
@rsodre i just pushed the majority of the tasks, its remaining the database which is to be discussed.
Also created a discord bot for testing purposes and the details are in the .env.example file so to test run for yourself, just copy it to .env.local file

**Comment by rsodre** [comment]
Looking good, thank you!

Our Supabase account has reached a limit, we'll decide tomorrow how to proceed

**Comment by Dprof-in-tech** [comment]
ok then.
could you assign me to another issue so I can continue building?

**Comment by Dprof-in-tech** [comment]
Hello Chief
I hope you're doing great.

I would try working on other issues on the repo while we try to resolve the
supabase thingy

**Comment by rsodre** [comment]
Ok, you would be good for #6, #7, #8, but now they all depend on #1 being ready
Looking at it right now

**Comment by Dprof-in-tech** [comment]
ok sure.
